clang/lib/Sema/SemaStmt.cpp

```diff
@@ -755,7 +755,7 @@ bool Sema::checkMustTailAttr(const Stmt *St, const Attr &MTA) {
     return true;
   };
 
-  const auto *CallerDecl = dyn_cast<FunctionDecl>(CurContext);
+  const auto *CallerDecl = getCurFunctionDecl();
```

Comment:
Unrelated?

Reply:
Stuff like this is from the implicit template region stuffs. There is another one like that that will likely fix that attr-tail.m test that was failing.